### ISSUE TITLE
Make PgAdmin, RedisCommander, and incoming MyPhpAdmin opt in per resource.

### ISSUE THREAD

**Issue by mitchdenny** [dotnet/aspire#2060]
Currently if we use `WithPgAdmin` (for example) on one resource, it opts all postgres resources in. It would be better to opt a particular resource into working with PgAdmin on a resource by resource basis.

**Comment by davidfowl** [comment]
Use an annotation to opt in?

**Comment by mitchdenny** [comment]
I'm actually thinking of creating a base annotation type. Something like this:

```csharp
public class RelationshipAnnotation<TSource, TDestination>(TSource source, TDestination) where TSource: IResource where TDestination: IResource
{
    public TSource Source { get; } = source;
    public TDestination Destination { get; } = destination;
}
```

The idea is that folks could sub class it - perhaps like this:

```csharp
public class PgAdminRegistrationAnnotation<PostgresDatabaseResource, PostgresServerREsource>(...) : RelationshipAnnotation(...)
{
}
```

The base class probably doesn't add much but perhaps we might be able to use this for things like expressing dependencies (its just a relationship after all).

**Comment by davidfowl** [comment]
Is it usable if it's generic? Can consuming code fill those type arguments?

**Comment by mitchdenny** [comment]
Yeah that is a good point. Non generic would be better for general purpose usage. The derivation might still be helpful for adding relationship data though.

**Comment by joperezr** [comment]
@mitchdenny are we doing this for P5? If so, is it okay if I assign this to you?

**Comment by davidfowl** [comment]
No, we should try to give this to someone else. @mitchdenny is focused on CDK work
